maintain black background styles between profiles

zelosleone opened the pull request:
#### Brief overview of PR changes/additions
Fixed issue where UI elements lose their black background when switching between profiles by:
- Adding per-profile style persistence 
- Implementing style caching and comparison
- Adding forced repaints for visual consistency
- Optimizing style application

#### Implementation Details
Added to Host class:
- `mProfileStyleSheet`: Stores profile-specific stylesheet
- `mWidgetStyleSheets`: Caches widget styles
- `reapplyStyles()`: Handles style persistence, made it fixed to avoid avoid redundant updates that could affect performance.

#### Testing
Tested with:
- Multiple StickMUD profiles
- Rapid profile switching
- Profile reconnections
- Application restart

#### Motivation for adding to Mudlet
Fixes UI consistency issue where background colors are lost when switching between profiles, particularly affecting StickMUD profiles.

#### Technical Changes
- Added style persistence per profile
- Implemented style comparison to prevent redundant updates
- Added forced repaints to fix rendering issues
- Optimized style application with caching

#### Other info (issues closed, discussion etc)
- Fixes #6561

zelosleone commented:
Not sure how tests failed, I can use the log files if you can send them to me.

vadi2 commented:
It looks like there is a build failure: https://github.com/Mudlet/Mudlet/actions/runs/12332274205/job/34419888473?pr=7558#step:21:688

vadi2 commented:
The PR is tough to review because it reformatted a lot of files - could you revert it or apply the [.clang-format](https://github.com/Mudlet/Mudlet/blob/development/src/.clang-format) style?

zelosleone commented:
Added signal to signals
`void signal_profileStylesChanged();
`
Added to Host.cpp:
```
void Host::reapplyStyles()
{
    static const QString BASE_STYLE = QStringLiteral("background-color: black;");

    if (mProfileStyleSheet == BASE_STYLE && 
        mWidgetStyleSheets.value("console") == BASE_STYLE && 
        mWidgetStyleSheets.value("userWindow") == BASE_STYLE && 
        mWidgetStyleSheets.value("label") == BASE_STYLE) {
        return;
    }

    mProfileStyleSheet = BASE_STYLE;

    if (mpConsole && mpConsole->styleSheet() != BASE_STYLE) {
        mpConsole->setStyleSheet(BASE_STYLE);
        mpConsole->repaint();
    }

    QMapIterator<QString, TConsole*> it(mConsoleMap);
    while (it.hasNext()) {
        it.next();
        if (it.value() && it.value()->styleSheet() != BASE_STYLE) {
            it.value()->setStyleSheet(BASE_STYLE);
            it.value()->repaint();
        }
    }

    mWidgetStyleSheets["console"] = BASE_STYLE;
    mWidgetStyleSheets["userWindow"] = BASE_STYLE;
    mWidgetStyleSheets["label"] = BASE_STYLE;

    emit signal_profileStylesChanged();
```

Added to QObject class:
```
class Host : public QObject
{
    Q_OBJECT

    friend class XMLexport;
    friend class XMLimport;
    friend class dlgProfilePreferences;

public:
    Host(int port, const QString& mHostName, const QString& login, const QString& pass, int host_id);
    ~Host();

    QString mProfileStyleSheet;
    QMap<QString, QString> mWidgetStyleSheets;
    void reapplyStyles();
```
